pygmt/accessors.py

```diff
@@ -95,3 +96,19 @@ def gtype(self, value):
                 f"Invalid coordinate system type: {value}, should be "
                 "either 0 for Cartesian or 1 for Geographic."
             )
+
+    def imshow(self):
+        """
+        Image plot of 2D DataArray.
+
+        Examples
+        --------
+
+        >>> from pygmt.datasets import load_earth_relief
+        >>> grid = load_earth_relief()
+        >>> grid.gmt.imshow()
+        """
+        fig = Figure()
+        fig.grdimage(self._obj, frame=True)
+        fig.colorbar()
+        fig.show()
```

Comment:
Debating on whether we should add `return fig`, so that users could potentially add extra stuff to the figure like so:

```python
fig = grid.gmt.imshow()
fig.colorbar()
fig.show()
```

or would that get a bit confusing? I'm thinking a bit about how matplotlib does it where users can add extra elements to the figure, but unsure if we want to mimic that in PyGMT.

Reply:
See my comment in https://github.com/GenericMappingTools/pygmt/issues/1412#issuecomment-1434341967.

Initially, I propose to add the `imshow()` function as a quick DataArray viewer, so I expect to view the grid using a simple single command.

Similar to the `ax` parameter in [`xarray.DataArray.plot.imshow`](https://xarray.pydata.org/en/v0.19.0/generated/xarray.DataArray.plot.imshow.html), I think the `imshow` function can at least accept a `fig` parameter.

If `fig=None` (the default behavior), `grid.gmt.imshow()` should plot the grid and show the image.

If a `Figure` instance is given, then we know that users want to plot the grid in an existing figure, then we should not show the image.

Then the `imshow()` will be like:
```python
    def imshow(self, fig=None):
        if fig is None:
            fig = Figure()

        fig.grdimage(self._obj, frame=True)
        fig.colorbar()

        if fig is None:
            fig.show()
        else:
            return fig
```